src/NServiceBus.Core/Audit/AuditProcessingStatisticsBehavior.cs

```diff
@@ -6,12 +6,13 @@
 
     class AuditProcessingStatisticsBehavior : IBehavior<IAuditContext, IAuditContext>
     {
+        internal const string ProcessingStartedKey = "NServiceBus.Auditing.ProcessingStarted";
+
         public Task Invoke(IAuditContext context, Func<IAuditContext, Task> next)
         {
-            if (context.Extensions.TryGet(out ProcessingStatisticsBehavior.State state))
```

Comment:
Wasn't the previous way of doing it more reliable with the stop watch because there is no issue of clock drift between the start and the end with that approach? This change would switch to setting the end no longer in relation to the start plus the time it took and is a behavior change in that regard.

Reply:
it would be more meaningful value but we've never read the stopwatch value. We could change this and move the entire logic into the fork-behavior and use the stopwatch approach there if we're happy to merge that logic with the fork behavior logic though.